### PR TITLE
Synchronize user newsletter subscription

### PR DESCRIPTION
**Pull request by ya-web**

### CODE REVIEW

File: config/services.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@
 # https://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration
 parameters:
     locale: 'fr'
+    mailchimp.list_id: '%env(NL_LIST_ID)%'
+    mailchimp.api_url: 'https://%env(API_SERVER)%.api.mailchimp.com/3.0/lists/%env(NL_LIST_ID)%/members'
```
Comment (kstefanini) [inline config/services.yaml:9]: `mailchimp.api_url: '%env(MAILCHIMP_API_URL)/lists/%env(NL_LIST_ID)%/members'`
With full root api url in `MAILCHIMP_API_URL`

File: config/services.yaml [hunk 2]
```diff
@@ -5,6 +5,10 @@
 # https://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration
 parameters:
     locale: 'fr'
+    mailchimp.list_id: '%env(NL_LIST_ID)%'
```
Comment (kstefanini) [inline config/services.yaml:8]: MAILCHIMP_LIST_ID

File: config/services.yaml [hunk 3]
```diff
@@ -5,6 +5,10 @@
...
+    mailchimp.api_url: 'https://%env(API_SERVER)%.api.mailchimp.com/3.0/lists/%env(NL_LIST_ID)%/members'
+    mailchimp.api_key: '%env(API_KEY)%'
```
Comment (kstefanini) [inline config/services.yaml:10]: MAILCHIMP_API_KEY

File: src/Controller/AdminController.php [hunk 4]
```diff
@@ -382,4 +405,33 @@ public function adminUserCancelDelete(
...
+            $data = $request->request->get('data');
+            if ($type && $data && $data['email']) {
+                /**
+                 * @var User $user
+                 */
+                $user = $manager->getRepository(User::class)
+                    ->findByEmail($data['email']);
+
+                if (!$user) {
+                    $logger->alert('unsubscribed user not found, check mailchimp audience');
```
Comment (kstefanini) [inline src/Controller/AdminController.php:429]: $logger->alert(sprintf('unsubscribed user with email %s not found, check mailchimp audience', $data['email']));

File: src/Service/MailchimpSyncContact.php [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+                $url,
+                $options
+            )
+                ->getContent();
+
+            $content = json_decode($jsonContent);
+            if (!empty($content) && !empty($content['status'])) {
+                return $content['status'];
+            }
+        } catch (TransportExceptionInterface | ClientExceptionInterface | RedirectionExceptionInterface | ServerExceptionInterface $e) {
```
Comment (kstefanini) [inline src/Service/MailchimpSyncContact.php:129]: If there is no specific catching for `(TransportExceptionInterface | ClientExceptionInterface | RedirectionExceptionInterface | ServerExceptionInterface` why don't you just catche `Exception`?

File: src/Service/MailchimpSyncContact.php [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+        $mailMessage = $this->twig->render('emails/newsletter-sync.html.twig', [
+            'user' => $user,
+            'type' => $type,
+        ]);
+
+        $this->mailer->send(
+            $user->getEmail(),
+            //'contact@obs-saisons.fr',
+            'idir.alliche.tb@gmail.com',
```
Comment (kstefanini) [inline src/Service/MailchimpSyncContact.php:159]: debug stuff ;)

File: src/Service/MailchimpSyncContact.php (hunk 5 above)
Comment (ya-web) [reply src/Service/MailchimpSyncContact.php:129]: PhpStorm advise but you're right

File: src/Service/MailchimpSyncContact.php [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+            'Un message a été envoyé à l’administrateur du site afin de régler le problème'
+        );
+
+        $mailMessage = $this->twig->render('emails/newsletter-sync.html.twig', [
+            'user' => $user,
+            'type' => $type,
+        ]);
+
+        $this->mailer->send(
+            $user->getEmail(),
```
Comment (kstefanini) [inline src/Service/MailchimpSyncContact.php:157]: `['contact@obs-saisons.fr', $user->getEmail()]`

File: src/Service/MailchimpSyncContact.php [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+            //'contact@obs-saisons.fr',
```
Comment (kstefanini) [inline src/Service/MailchimpSyncContact.php:158]: To: `['contact@obs-saisons.fr', $user->getEmail()]]` both user and admin

File: src/Service/MailchimpSyncContact.php [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+            'idir.alliche.tb@gmail.com',
+            $this->mailer->getSubjectFromTitle($mailMessage),
+            $mailMessage
+        );
+
+        $user->setIsNewsletterSubscriber($isSubscription);
```
Comment (kstefanini) [inline src/Service/MailchimpSyncContact.php:164]: Why do you set that here?

Comment (ya-web) [reply src/Service/MailchimpSyncContact.php:164]: oh sorry I was meaning `$user->setIsNewsletterSubscriber(!$isSubscription);`

Comment (ya-web) [reply src/Service/MailchimpSyncContact.php:164]: if subscription or unsubscription didn't work you may want to roll it back in database too

**Pull request merged**